framework/JobHandler.py

```diff
@@ -401,20 +418,29 @@ def isFinished(self):
       @ In, None
       @ Out, isFinished, bool, True all the runs in the queue are finished
     """
+    tempList=copy.copy(self.__running+self.__clientRunning)
+    len1 = copy.deepcopy(len(self.__queue))
+    len2 = copy.deepcopy(len(self.__clientQueue))
```

Comment:
Do you need these copy and deepcopies?

Reply:
This is has been cleaned as it is now in devel